WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -225,8 +225,9 @@ def post_results
         body += "#{record_strs.join(", ")}.  \n" # Trailing spaces for markdown give us a <br>
       end
     end
-    post = Post.new(title: title, body: body, author: current_user, world_readable: true)
-    post_post(post)
+    comp.update(results_posted_at: Time.now)
+    comp.save!
```

Comment:
I'd rather the user get an ugly expection page and know something didn't work than have this silently fail to do anything.

Posting results is not an atomic event. It's a manual process that involves running several scripts, and some manual eyeballing of the data before the results team member posting the results decides "okay, this is good, let's tell the world about this". That moment is when the announcement is created, and I think is the appropriate time to set the `results_posted_at` field. An artifact of this design is that it's possible for civilians to see results coming in as the results team is working on uploading results. A potential later improvement would be for us to only display results to the public if their competition has `results_posted_at` set to true.

A better name for this method might be `create_results_announcement_and_redirect`.